src/crate/client/http.py

```diff
@@ -258,6 +283,28 @@ def _create_sql_payload(stmt, args, bulk_args):
     return json.dumps(data, cls=CrateJsonEncoder)
 
 
+def _get_socket_opts(keepalive=True,
+                     tcp_keepidle=None,
+                     tcp_keepintvl=None,
+                     tcp_keepcnt=None):
+    """
+    Return an optional list of socket options for urllib3's HTTPConnection
+    constructor.
+    """
+    if not keepalive:
+        return None
+
+    opts = [(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)]
```

Comment:
Just a small nit to improve inline documentation about what we are essentially doing here.

```suggestion
    # Let's always use TCP keepalive.
    opts = [(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)]

    # Optionally adjust TCP keepalive parameters.
```

Reply:
Not so sure if this improves things. It repeats what the code already tells a reader, but doesn't give a rationale. ("Document *why*, not *what*")

Something like this would imho make more sense.

```suggestion
    # Prevents issues where load balancers or ISPs terminate in-activate connections
    # See https://joonas.fi/2017/01/23/microsoft-azures-networking-is-fundamentally-broken/ 
    opts = [(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)]
```